File: Quarkus/code-with-quarkus/src/main/java/org/acme/DTOs/EventDTO.java
package org.acme.DTOs;

import java.time.LocalDateTime;
import java.util.List;

public class EventDTO {
    private Long id;
    private String title;
    private String description;
    private int basicSeats;
    private int premiumSeats;
    private int vipSeats;
    private int avBasicSeats;
    private int avPremiumSeats;
    private int avVipSeats;
    private LocalDateTime eventDateTime;
    private Long etypeId;
    private OrganizerDTO organizer;
    private List<SellerDTO> sellers;
    private List<TicketDTO> tickets;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getBasicSeats() {
        return basicSeats;
    }

    public void setBasicSeats(int basicSeats) {
        this.basicSeats = basicSeats;
    }

    public int getPremiumSeats() {
        return premiumSeats;
    }

    public void setPremiumSeats(int premiumSeats) {
        this.premiumSeats = premiumSeats;
    }

    public int getVipSeats() {
        return vipSeats;
    }

    public void setVipSeats(int vipSeats) {
        this.vipSeats = vipSeats;
    }

    public int getAvBasicSeats() {
        return avBasicSeats;
    }

    public void setAvBasicSeats(int avBasicSeats) {
        this.avBasicSeats = avBasicSeats;
    }

    public int getAvPremiumSeats() {
        return avPremiumSeats;
    }

    public void setAvPremiumSeats(int avPremiumSeats) {
        this.avPremiumSeats = avPremiumSeats;
    }

    public int getAvVipSeats() {
        return avVipSeats;
    }

    public void setAvVipSeats(int avVipSeats) {
        this.avVipSeats = avVipSeats;
    }

    public LocalDateTime getEventDateTime() {
        return eventDateTime;
    }

    public void setEventDateTime(LocalDateTime eventDateTime) {
        this.eventDateTime = eventDateTime;
    }

    public Long getETypeId() {
        return etypeId;
    }

    public void setETypeId(Long eTypeId) {
        this.etypeId = eTypeId;
    }

    public OrganizerDTO getOrganizer() {
        return organizer;
    }

    public void setOrganizer(OrganizerDTO organizer) {
        this.organizer = organizer;
    }

    public List<SellerDTO> getSellers() {
        return sellers;
    }

    public void setSellers(List<SellerDTO> sellers) {
        this.sellers = sellers;
    }

    public List<TicketDTO> getTickets() {
        return tickets;
    }

    public void setTickets(List<TicketDTO> tickets) {
        this.tickets = tickets;
    }
}

File: Kotlin/app/src/main/java/com/example/ticketmag/DTOs/EventDTO.kt
package com.example.ticketmag.DTOs

import kotlinx.serialization.Serializable
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class EventDTO(
    val id: Long? = null,
    val title: String? = null,
    val description: String? = null,
    val basicSeats: Int = 0,
    val premiumSeats: Int = 0,
    val vipSeats: Int = 0,
    val avBasicSeats: Int = 0,
    val avPremiumSeats: Int = 0,
    val avVipSeats: Int = 0,
    val eventDateTime: LocalDateTime? = null,
    val etypeId: Long? = null,
    val organizer: OrganizerDTO? = null,
    val sellers: List<SellerDTO>? = null,
    val tickets: List<TicketDTO>? = null
) {

    private fun parseEventDateTime(): LocalDateTime? =
        eventDateTime?.let { LocalDateTime.parse(it.toString(), DateTimeFormatter.ISO_LOCAL_DATE_TIME) }

    fun getFormattedDate(): String =
        parseEventDateTime()?.format(DateTimeFormatter.ISO_LOCAL_DATE) ?: "No Date"

    fun getFormattedTime(): String =
        parseEventDateTime()?.format(DateTimeFormatter.ofPattern("HH:mm")) ?: "No Time"
}